replica/table.cc

```diff
@@ -1726,23 +1719,23 @@ table::make_partition_presence_checker(const sstables::sstable_set& ssts) {
 snapshot_source
 table::sstables_as_snapshot_source() {
     return snapshot_source([this] () {
-        auto sst_set = _sstables;
+        auto sst_set = make_lw_shared<sstable_set>(_sstables.clone());
```

Comment:
We moved refresh_compound_sstable_set() here, which is still not exception safe, because in row cache we need to take the snapshot atomically with execute():

```
    [&] () noexcept {
        _prev_snapshot_pos = dht::ring_position::min();
        _prev_snapshot = std::exchange(_underlying, _snapshot_source());
```